template/Module/src/Module.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace %name%;
+
+use DirectoryIterator;
+
+class Module
+{
+    public function getConfig() : array
+    {
+        $config = [];
+
+        foreach (new DirectoryIterator(__DIR__ . '/../config') as $fileInfo) {
```

Comment:
This is new approach, we haven't used before.
What I am proposing here is to have a `config` directory in module and "small" separate files for "each section". So the section name is the file name, and the content is the value.
Then we should finish up with:
```
config/controllers.php - configuration for controllers (controller plugin manager)
config/service_manager.php - DI configuration
config/laminas-cli.php - Laminas CLI configuration
config/view_manager.php - view configuration
config/router.php - router configuration
...
```

If we  have something like that it should be easier to register controller/factory, create routing etc... within the module, instead of "hacking" into `Module.php`. I know this is new way, and we haven't never suggested anything like that, but with laminas-cli, as a new feature, we can provide new solutions.

In commands we can detect easily that there is no file `controllers.php` and we can create it, and suggest adding it to `Module::getConfig` (if module is using the old style configuration) by adding:
```php
'controllers' => __DIR__ . '/../config/controllers.php',
```

Reply:
I used the same in my applications for _smaller_ files with better overview like `config/navigation.php`.

Reply:
Another possibility is to use `module.config.php` like we did before... but have _that_ file include in the various other config files it needs in order to define the full module configuration.

Honestly, either approach works for me. If we do this, however, we should document it heavily, so people know it's even possible.

Reply:
I was thinking about extracting it somewhere - either trait or 'abstract config module'.